### PR TITLE
feat: add new SequencerInbox getters

### PR DESCRIPTION
**Pull request by chrstph-dvx**
Part of FS-535

Getters:
https://github.com/OffchainLabs/arbitrum-orbit-sdk/pull/144
https://github.com/OffchainLabs/arbitrum-orbit-sdk/pull/145
https://github.com/OffchainLabs/arbitrum-orbit-sdk/pull/146
https://github.com/OffchainLabs/arbitrum-orbit-sdk/pull/147

Setters:
https://github.com/OffchainLabs/arbitrum-orbit-sdk/pull/148
https://github.com/OffchainLabs/arbitrum-orbit-sdk/pull/149
https://github.com/OffchainLabs/arbitrum-orbit-sdk/pull/150

### CODE REVIEW

File: src/actions/getMaxTimeVariation.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+export type GetMaxTimeVariationReturnType = {
+  delayBlocks: bigint;
+  futureBlocks: bigint;
+  delaySeconds: bigint;
+  futureSeconds: bigint;
+};
+
+export async function getMaxTimeVariation<TChain extends Chain | undefined>(
+  client: PublicClient<Transport, TChain>,
+  args: GetMaxTimeVariationParameters<false>,
```
Comment (fionnachan) [inline src/actions/getMaxTimeVariation.ts:20]: does this need `<false>`?

File: src/actions/isBatchPoster.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+type Args = {
+  batchPoster: Address;
+};
+type SequencerInboxABI = typeof sequencerInbox.abi;
+export type IsBatchPosterParameters<Curried extends boolean = false> = ActionParameters<
+  Args,
+  'sequencerInbox',
+  Curried
+>;
```
Comment (fionnachan) [inline src/actions/isBatchPoster.ts:13]: this is very similar to `GetMaxTimeVariationParameters` and `IsValidKeysetHashParameters` etc

do we really want an independent one in every file?

Comment (chrstph-dvx) [reply src/actions/isBatchPoster.ts:13]: Yes, you can see the reasoning here: https://linear.app/offchain-labs/issue/FS-504/explore-ways-to-simplify-contract-specific-utilities-for-v1#comment-6ca0cf13

File: src/actions/getMaxTimeVariation.ts (hunk 1 above)
Comment (chrstph-dvx) [reply src/actions/getMaxTimeVariation.ts]: good point, we don't

File: src/types/Actions.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ */
+export type ActionParameters<Args, ContractName extends string, Curried extends boolean> = Prettify<
+  Curried extends false
+    ? Args & { [key in ContractName]: Address }
+    : Args extends Record<string, never>
+    ?
+        | {
+            [key in ContractName]: Address;
+          }
+        | void
```
Comment (fionnachan) [inline src/types/Actions.ts:18]: continuing our previous discussion, not very sure about using `void` for optional objects but approving this as it works

**Review by fionnachan — APPROVED** [review]

File: src/types/Actions.ts (hunk 3 above)
Comment (chrstph-dvx) [reply src/types/Actions.ts:18]: It's slightly different here. If we use `void, we allow `getMaxTimeVariation()` if the sequencerInbox was passed to the decorator previously. If we use `{ [key in ContractName]?: Address }` instead, then we always require a parameter, also `getMaxTimeVariation({ sequencerInbox: undefined })` would now be a valid option.

**Pull request merged**